File: code-smells-project/src/middlewares/error_handler.py
"""Tratamento de erros centralizado.

Substitui os blocos `try/except Exception` espalhados pelos controllers. Erros
inesperados são logados internamente e respondidos com mensagem genérica, sem
vazar detalhes de implementação ao cliente.
"""
import logging

from flask import jsonify

logger = logging.getLogger(__name__)


def register_error_handlers(app) -> None:
    @app.errorhandler(404)
    def handle_not_found(_error):
        return jsonify({"erro": "Recurso não encontrado"}), 404

    @app.errorhandler(Exception)
    def handle_unexpected(error):
        logger.exception("Erro não tratado: %s", error)
        return jsonify({"erro": "Erro interno"}), 500

File: ecommerce-api-legacy/src/middlewares/errorHandler.js
const AppError = require('../errors/AppError');
const logger = require('../utils/logger');

// Tratamento de erro centralizado. Erros de domínio (AppError) preservam o
// status e a mensagem originais; erros inesperados são logados e respondem 500
// genérico, sem vazar stack trace ao cliente.
function errorHandler(err, req, res, next) {
    if (err instanceof AppError) {
        return res.status(err.status).send(err.message);
    }

    logger.error('unhandled_error', { path: req.path, message: err.message });
    return res.status(500).send('Erro interno');
}

module.exports = errorHandler;
